src/software/ai/evaluation/q_learning/attacker_mdp_feature_extractor.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include "software/ai/evaluation/q_learning/attacker_mdp_action.h"
+#include "software/ai/evaluation/q_learning/feature_extractor.hpp"
+
+/**
+ * State representation of the Markov decision process (MDP) modelling
+ * the Attacker agent's gameplay decision making.
+ */
+struct AttackerMdpState
+{
+    WorldPtr world_ptr;
+    std::shared_ptr<Strategy> strategy;
+};
+
+/**
+ * FeatureExtractor that extracts the values of features from a AttackerMdpState.
+ */
+class AttackerMdpFeatureExtractor
+    : public FeatureExtractor<AttackerMdpState, AttackerMdpAction>
+{
+   public:
+    explicit AttackerMdpFeatureExtractor();
```

Comment:
the docs for the constructor should describe the order of features initialized (it'll help understand the ordering of features when we look at the Q-Learning widget)

Reply:
added feature and actions names to the Q-learning widget, probably don't need to add docs any more